Add native `<input type="file">` styles

lindsaym-fa opened the pull request:
Closes #1241 

Adds styles and documentation for `<input type="file">`

**Safari**
<img width="278" height="97" alt="Screenshot 2025-08-07 at 11 20 30 AM" src="https://github.com/user-attachments/assets/18e2fc42-57f7-4920-b5f6-425ab35c133f" /><img width="353" height="97" alt="Screenshot 2025-08-07 at 11 20 40 AM" src="https://github.com/user-attachments/assets/f32bd6ec-04f3-4d48-bf0f-6522a14adf24" />


**Firefox**
<img width="263" height="97" alt="Screenshot 2025-08-07 at 11 18 40 AM" src="https://github.com/user-attachments/assets/fcdf181d-c609-4d3f-ba43-2bfc6bb4ad6d" /><img width="391" height="97" alt="Screenshot 2025-08-07 at 11 37 40 AM" src="https://github.com/user-attachments/assets/5c1462c8-6f19-4f7b-b5cb-81321e85a0dc" />


**Chrome**
<img width="263" height="97" alt="Screenshot 2025-08-07 at 11 18 53 AM" src="https://github.com/user-attachments/assets/fde8a2a4-b0a6-4ab7-b9bd-ab87440d095d" /><img width="359" height="97" alt="Screenshot 2025-08-07 at 11 43 30 AM" src="https://github.com/user-attachments/assets/47b6ca1c-8fd8-4076-b77e-1ac918424c7b" />

lindsaym-fa committed:
add `input type="file"` styles

lindsaym-fa committed:
add `max-inline-size`

claviska reviewed (dismissed):
This looks great!

It looks like we have some `.wa-` modifier classes in native styles for buttons. Should we adapt this to file input buttons as well?

claviska commented:
Needs a changelog entry :)

lindsaym-fa committed:
refactor button styles to include `::file-selector-button`

lindsaym-fa committed:
improve docs

lindsaym-fa committed:
add changelog

lindsaym-fa commented:
Great calls, thank you!

Added a changelog entry, refactored the styles to leverage the existing modifier classes (and serendipitously fixed a bug with missing `:hover` and `:active` styles for native buttons without a modifier class), and improved the docs.

lindsaym-fa committed:
add region marker
